src/watch-run.ts

```diff
@@ -37,26 +45,56 @@ export function makeWatchRun(instance: TSInstance) {
           filePath.match(constants.dtsDtsxOrDtsDtsxMapRegex) !== null &&
           filePath.match(constants.nodeModules) === null
         ) {
-          updateFile(instance, filePath);
+          promises.push(updateFile(instance, filePath, loader, loaderIndex));
         }
       }
     }
 
     // Update all the watched files from solution builder
     if (instance.solutionBuilderHost) {
       for (const filePath of instance.solutionBuilderHost.watchedFiles.keys()) {
-        updateFile(instance, filePath);
+        promises.push(updateFile(instance, filePath, loader, loaderIndex));
       }
     }
 
-    callback();
+    Promise.all(promises)
+      .then(() => callback())
+      .catch(err => callback(err));
   };
 }
 
-function updateFile(instance: TSInstance, filePath: string) {
-  updateFileWithText(
-    instance,
-    filePath,
-    nFilePath => readFile(nFilePath) || ''
-  );
+function updateFile(
+  instance: TSInstance,
+  filePath: string,
+  loader: webpack.loader.LoaderContext,
+  loaderIndex: number
+) {
+  return new Promise<void>((resolve, reject) => {
+    if (
+      loaderIndex + 1 < loader.loaders.length &&
```

Comment:
Could you put a comment here to clarify the purpose of the 2 code branches here please?

Reply:
OK, I will.